### ISSUE TITLE
Using custom extensions on windows fails with ananconda (_run_node, compile.js, communicate, hangs)

### ISSUE THREAD

**Issue by PierreBizouard** [bokeh/bokeh#5740]
When running examples (ionrangeslider, latex legend etc.) with custom __implementation__ on my windows 10/anaconda 32 bit setup, bokeh hangs. It hangs in subprocess.py _communicate, the starting of reader threads fails.

* bokeh info : Python version      :  3.4.3 |Continuum Analytics, Inc.| (default, Dec  1 2015, 11:59:33) [MSC v.1600 32 bit (Intel)]
IPython version     :  4.1.2
Bokeh version       :  0.12.4
BokehJS static path :  C:\Anaconda3\envs\py34\lib\site-packages\bokeh\server\static

a quick and dirty workaround in bokeh.util.compiler.py :

```python

fix_win = True if "win" in sys.platform else False

def _run(app, argv, input=None, timeout=10):
    if _nodejs is None:
        raise RuntimeError('node.js is needed to allow compilation of custom models ' +
                           '("conda install -c bokeh nodejs" or follow https://nodejs.org/en/download/)')
    input_data = None if input is None else json.dumps(input).encode()
    proc = Popen([app] + argv, stdout=PIPE, stderr=PIPE, stdin=PIPE)
    if fix_win:
        proc.stdin.write(input_data)
        proc.stdin.close()
        proc.wait(timeout=timeout)
        stdout = proc.stdout.read()
        errout = proc.stderr.read()
    else:
        stdout, errout = proc.communicate(input=input_data, timeout=10)

    if proc.returncode != 0:
        raise RuntimeError(errout)
    else:
        return stdout.decode()
```

bokeh is an awesome work congratulation to the team

**Comment by bryevdv** [comment]
@PierreBizouard I apologize that this did not get triaged earlier, that is why it got overlooked. Can you comment if this is still an issue on current versions? If so can you provide a complete test script for us to work with?

**Comment by mattpap** [comment]
Just to note, I recall this kind of issue, i.e. improper usage of `subprocess` module, being fixed at some point. Though I'm not sure if this covered windows.

**Comment by bryevdv** [comment]
Seems to have been fixed at some point (`0.12.16dev4):

<img width="997" alt="screen shot 2018-05-11 at 11 32 30" src="https://user-images.githubusercontent.com/1078448/39940712-09a156ba-550f-11e8-9dc7-58b22e388ec7.png">
